src/main/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGenerator.java

```diff
@@ -27,7 +27,7 @@
      */
     public static final String APPENDIX_CHARACTERS = "abcdefghijklmnopqrstuvwxyz";
     private static final Logger LOGGER = LoggerFactory.getLogger(BibtexKeyGenerator.class);
-    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"-#~^:'`ʹ";
+    private static final String KEY_ILLEGAL_CHARACTERS = "{}(),\\\"#~^:'`ʹ";
     private static final String KEY_UNWANTED_CHARACTERS = "{}(),\\\"-";
```

Comment:
Why didn't you remove the `-` it here?

Reply:
According to the recommendation in the [README](https://github.com/JabRef/user-documentation/blob/master/en/fields/README.md), I thought it was a mistake to include a dash in illegal signs, not those not recommended.

> Some characters should not be used in bibtexkey as they are not compatible or not recommended: `{ } ( ) , \ " - # ~ ^ : '`

Reply:
@koppor so should i remove dash from unwanted characters and make update on READEME file?

Reply:
Remove it from illegal, but keep it in unwanted.

Reply:
I checked the usage of `KEY_UNWANTED_CHARACTERS`.

The code comment reads as follows:

```java
// User doesn't want us to enforce legal characters. We must still look
// for whitespace and some characters such as commas, since these would
// interfere with parsing:
```

Thus, the `KEY_UNWANTED_CHARACTERS` clearly is a subset of `KEY_ILLEGAL_CHARACTERS`. Not sure, whether `KEY_ILLEGAL_CHARACTERS` makes sense at all. Maybe because of tool interopability.

I see three options:

A) Remove `KEY_UNWANTED_CHARACTERS` completely - and also the preference for it.
B) Set `KEY_UNWANTED_CHARACTERS` to a minimum - as indicated in the comment. I would propose `{}"` since these are the characters causing issues at parsing. I would not add space since this did not cause any issues until now (at list that I am aware of)
C) Just remove `-` from `KEY_UNWANTED_CHARACTERS`

Please do option B). Please rename `KEY_UNWANTED_CHARACTERS` to `KEY_CHARACTERS_CAUSING_PARSING_ERRORS`. Please add a comment above that this is a subset of `KEY_ILLEGAL_CHARACTERS`.